immune_subtype_clustering/prepare_data_sheet.py

```diff
@@ -0,0 +1,99 @@
+import os
+import pandas as pd
+import synapseclient
+from synapseclient import File
+from typing import List
+
+
+def syn_login():
+    syn = synapseclient.Synapse()
+    syn.login()
+    return syn
+
+
+def download_data_files(parent: str):
+    """downloads all TSV files in parent synapse folder from synapse id provided"""
+    syn = syn_login()
+    children = syn.getChildren(parent=parent)
+    syn_id_list = []
+    for child in children:
+        if str(child["name"]).endswith(".tsv"):
+            syn_id_list.append(child["id"])
+            syn.get(str(child["id"]), downloadLocation="./data")
+    return syn_id_list
+
+
+def load_data_files(data_directory: str) -> List[pd.DataFrame]:
+    """loads all files in data_diectory into dataframes, creates Hugo column,
+    grabs Hugo and tpm columns, renames tppm column to patient id, appends each df to gene_df_list
+    """
+    data_directory = "./data"
+    files = [f for f in os.listdir(data_directory)]
+    print(f"Files: {files} retrieved from data_directory")
+    gene_df_list = []
+    for file in files:
+        patient_id = file.split("_")[0]
+        df = pd.read_csv(data_directory + "/" + file, sep="\t")
+        df.insert(0, "Hugo", df["target_id"].str.split("|", expand=True)[5])
+        gene_df = df[["Hugo", "tpm"]]
+        gene_df = gene_df.rename(columns={"tpm": patient_id})
+        gene_df_list.append(gene_df)
+        print(f"Data from {file} loaded")
+    return gene_df_list
+
+
+def merge_dfs(gene_df_list: List[pd.DataFrame]) -> pd.DataFrame:
+    """Joins all dfs in gene_df_list by index, renames all Hugo columns after first by index, returns final_df"""
+    final_df = gene_df_list[0]
+    gene_df_list.pop(0)
+    for i, gene_df in enumerate(gene_df_list):
+        gene_df = gene_df.rename(columns={"Hugo": f"Hugo_{i}"})
```

Comment:
What is your rationale for doing this rename?

Reply:
You can't merge data frames with more than two of the same column names (they all have a "Hugo" column") when merging by index. I am merging by index because there are duplicates in the "Hugo" columns, so merging on the "Hugo" column results in a massive expansion of the number of rows. When you merge dataframes with duplicate columns not on those columns, you get a merged dataframe with "Hugo_x" and "Hugo_y" and then the next merge results in this error:
```
MergeError: Passing 'suffixes' which cause duplicate columns {'Hugo_x'} is not allowed.
```

Reply:
Ah that makes sense. Just confirming that you're saying that if you merge on the 'Hugo' column instead of index, it creates extra rows for the duplicates?

If so, can we add a comment?

Reply:
Right. Because there are duplicates in the Hugo column, the merge doesn't know which rows to line up. I will add a comment.